crates/sidecar/src/main.rs

```diff
@@ -28,14 +28,38 @@ async fn run(
 ) -> anyhow::Result<()> {
     let connection_info = ConnectionInfo::from_path(connection_file_path).await?;
 
-    let mut iopub_connection = connection_info
-        .create_client_iopub_connection("", "sidecar-session")
-        .await?; // todo: generate session ID
+    let mut iopub = connection_info
+        .create_client_iopub_connection("", &format!("sidecar-{}", uuid::Uuid::new_v4()))
+        .await?;
+
+    let mut shell = connection_info
+        .create_client_shell_connection(&iopub.session_id)
+        .await?;
+
+    let (tx, mut rx) = futures::channel::mpsc::channel::<JupyterMessage>(100);
 
-    let _webview = WebViewBuilder::new(&window)
+    smol::spawn(async move {
+        while let Some(message) = rx.next().await {
+            if let Err(e) = shell.send(message).await {
+                eprintln!("Failed to send message: {}", e);
+                break;
+            }
+        }
+    })
+    .detach();
+
+    let webview = WebViewBuilder::new(&window)
         .with_devtools(true)
-        .with_asynchronous_custom_protocol("sidecar".into(), move |request, responder| {
-            let response = get_response(request).map_err(|e| {
+        .with_asynchronous_custom_protocol("sidecar".into(), move |req, responder| {
+            if let (&Method::POST, "/message") = (req.method(), req.uri().path()) {
```

Comment:
I figured we could use IPC... or just have an endpoint for our custom protocol. I guess the nice thing with this is our data is richer (we can use a Response, which could have binary data too).